CRUD-API/lib/DataTypes/product/index.js

```diff
@@ -28,40 +28,58 @@ class Product {
       productName,
       productDescription
    ) {
-      if (
-         getType(productNumber) == types.Number &&
-         getType(cost) == types.Number &&
-         getType(markup) == types.Number &&
-         getType(sellAtList) == types.Boolean &&
-         getType(productName) == types.String &&
-         getType(productDescription) == types.String
-      ) {
-         this.number = productNumber;
-         this.cost = cost;
-         this.markup = markup >= this.MIN_MARKUP ? markup : this.MIN_MARKUP;
-         this.listPrice = listPrice;
-         this.sellAtList = sellAtList;
-         this.productName = productName;
-         this.productDescription = productDescription;
-      } else {
-         console.error(
-            'Invalid arguments.',
-            productNumber,
-            cost,
-            markup,
-            listPrice,
-            sellAtList,
-            productName,
-            productDescription
-         );
+      switch (true) {
```

Comment:
I'm curious as to the reasoning behind a switch statement here. Makes readability a little challenging.

Reply:
The `switch (true)` approach allows for a consistent syntax when dealing with multiple conditional checks. e.g (`   
         getType(productNumber) == types.Number &&
            getType(cost) == types.Number &&
            getType(markup) == types.Number &&
            getType(sellAtList) == types.Boolean &&
            getType(productName) == types.String &&
            getType(productDescription) == types.String)` are cases which are supposed to be met if i'm right , using a switch was to avoid multiple if-else  blocks



1. Here's an explanation of how `switch(true)` is used in this context:

`switch(true)`:

- In a traditional `switch` statement, you would typically provide a value to be compared against different cases. For example, `switch(variable)` where `variable` is the value to be evaluated against various cases.

-  However, in this code, the `switch` statement uses the constant value `true`
      switch (true) {
This means that the code will enter the first case block whose condition evaluates to `true`.


2. case Statement:

- Within the `switch(true)` block, there is a single `case` statement with a condition.

- The condition is a complex expression that involves multiple calls to `getType()` and comparisons with `types` constants. This condition is evaluating the types of various parameters.

3. Handling the Valid Case:

- If the condition in the `case` statement evaluates to `true`, the code inside the `case` block is executed.

- In this case, it initializes various properties of the `Product` class (e.g., `this.number`, `this.cost`, etc.).

- It also performs an additional check related to `this.overrideCheck` and logs an error message if necessary.

4. Default Case:

If none of the conditions in the `case` statements are met, the code falls into the `default` case.
In this case, it logs an error message indicating that the provided arguments are invalid.

Currently I'm using an if statement in replace of the switch statement

Reply:
I understand perfectly the use of switch here. I know it will work. My gripe is not with that. It's that this is not a proper use case for switch, as switch is for multiple cases. Switching on 'true' is considered bad form and does affect the readability. I know that there is a boatload of conditionals, but that's the problem with JavaScript -- there is no type enforcement on function/method/constructor calls. Another thing I do is, sometimes I create a private static class method 'validateArguments' in which I handle all of the argument validation (if it's more than a simple type check). In the case of simple type checking, I just opt for if/else because that's the best tool for the job. 
if true -> proceeded with the construction 
else -> error 

That's cleaner than a switch. Though a switch will totally work, I much prefer if/else. 

To be fair on this call, I spoke to a former colleague and asked him to review and he agreed.